### PR TITLE
Docs: add Docker docs

### LINKED ISSUE #1667: Docker Deployment docs
I've noticed the docs prioritize Astro deployment with serverless (or static) hosts. I personally use Docker for deploying things on non-serverless hosts, so I feel others should be able to do the same without having to do any research.

I can easily make a tutorial and make a PR.

LMK if anyone has any questions!

### PR DESCRIPTION
**Pull request by ElianCodes**
#### What kind of changes does this PR include?

- New or updated content

#### Description

This adds a (temporary) Docker page to the deployment guides.

Later on, we should find a different place (recipes?) for this, since it's not really a "Deployment guide".
We should also change up the content of the GCP deployment a little since half of it is duplicated content. For now I think it's best to get this merge and then focus on making the GCP deployment guide compatible with the Docker guide.

- Closes #1667

### CODE REVIEW

**Comment by sarah11918** [comment]
Thanks for this, @ElianCodes !  :partying_face: 

We are actually *very close* to launching an official recipes section. (See PR #2550 ) What do you think about including this there, with our Recipes launch? @Jutanium , thoughts?

**Comment by Jutanium** [comment]
Agreed - we can edit it to fit a bit more with the recipe style, and I think this will work great.

**Comment by sarah11918** [comment]
OK, so if a recipe, then we would really only need the file `src/content/docs/recipes/docker.mdx`.

@Jutanium: how are you currently working on recipes now? Are you working directly on the reorg branch with new recipes, or are you just writing content for now?  Can you let @ElianCodes know how to proceed with working on this as a recipe?

**Comment by ElianCodes** [comment]
Noticed that the Recipes section is online and merged! 🥳 Congrats on that!

Want me to go ahead and fix this to fit in @Jutanium ?

**Comment by ElianCodes** [comment]
Allright! I'll get that done!

Thanks @sarah11918

**Comment by ElianCodes** [comment]
Just checked your changes locally @Jutanium ! They look great 👌

File: src/content/docs/en/recipes/docker.mdx [hunk 1]
```diff
@@ -0,0 +1,137 @@
+---
+title: Build your Astro Site with Docker
+description: How to build your Astro site to the web using Docker.
```
Comment (ElianCodes) [inline src/content/docs/en/recipes/docker.mdx:3]: Shouldn't this be "How to build & deploy your Astro site to the web using Docker"?

Comment (Jutanium) [reply src/content/docs/en/recipes/docker.mdx]: Changed to "build your Astro site using Docker" since we don't teach how to deploy

**Comment by sarah11918** [comment]
OK... 105 files changed?  :sweat_smile: 

Given that the nature of this PR has changed, and like ALL of Docs has been pulled into it... can I respectfully ask that we close THIS PR, and the ONE page of content we need (doesn't need any infra/logo etc. now) is PR'd to the now-existing recipe location?  :smile:  Would that work for people?

**Comment by ElianCodes** [comment]
**👀 Me looking angry at Dan**

Just kidding, I'll do that in a few minutes! You're totally right @sarah11918 😅

**Comment by ElianCodes** [comment]
@sarah11918, I've replaced this PR by https://github.com/withastro/docs/pull/2815

**Pull request closed without merging**